src/main/java/seedu/student/model/appointment/SameDateAppointmentList.java

```diff
@@ -88,12 +91,26 @@ public void add(Appointment toAdd) {
     }
 
     /**
-     * Replaces the student {@code target} in the list with {@code editedStudent}.
+     * Replaces the appointment {@code target} in the list with {@code editedAppointment}.
      * {@code target} must exist in the list.
-     * The student identity of {@code editedStudent} must not be the same as another existing student in the list.
+     * The appointment identity of {@code editedAppointment} must not be the same as another existing appointment
+     * in the list.
      */
-    public void setAppointment(Appointment target, Appointment editedAppointment) {
-        // TODO
+    public void setAppointment(Appointment target, Appointment editedAppointment) throws CommandException {
+        requireAllNonNull(target, editedAppointment);
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            //throw exception here?
+            throw new CommandException(MESSAGE_NONEXISTENT_APPOINTMENT);
+        }
```

Comment:
Correct me if I'm wrong, shouldn't `CommandException` be thrown only in the `Logic` component?

Reply:
Noted, I'm now using an assertion for defensive purposes instead of throwing an exception